.github/workflows/ci-protobuf.yml

```diff
@@ -0,0 +1,47 @@
+# Continuous integration handling for Protocol Buffers
+name: ci-protobuf
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/ci-protobuf.yml
+      - .go-version
+      - go.mod
+      - go.sum
+      - 'tfprotov5/internal/tfplugin5/*'
+      - 'tfprotov6/internal/tfplugin6/*'
+
+permissions:
+  contents: read
+
+jobs:
+  protoc:
```

Comment:
So, this is all good, but I think we should add a couple of things so that the knowledge isn't so "buried".

I was looking for a `Makefile` in the project to help doing something like `make generate` (to take care of the code generation). And there is no explanation about this bit of the development workflow in the `README`.

So I dug up the `generate.sh` and found:

```
# We do not run protoc under go:generate because we want to ensure that all
# dependencies of go:generate are "go get"-able for general dev environment
# usability. To compile all protobuf files in this repository, run
# "make protobuf" at the top-level.
```

So, I suspect this is a rethinking of the original approach (that I frankly don't fully understand because it wasn't explained in the `generate.sh` either).

Also, that bit above mentions a `Makefile` that doesn't seem to exist anymore.

I don't come from an "`Makefile`-heavy" background, but I think it would be preferable if all "developer's default actions" (not sure how to best describe it) were ready made in there. Compared to the README, that is textual description, the `Makefile` helps on a more "don't tell me, show me" level.

What do you think?

Reply:
As a matter of fact, I'd argue that the `generate.sh` scripts should just be replaced with `Makefile` commands, given that all the script does is:

- figure out its own directory
- call `protoc`